pypika/tests/test_query.py

```diff
@@ -204,3 +204,29 @@ def test_query_builders_have_reference_to_correct_query_class(self):
 
         with self.subTest('OracleQueryBuilder'):
             self.assertEqual(OracleQuery, OracleQueryBuilder.QUERY_CLS)
+
+    def test_pipe(self) -> None:
+        base_query = Query.from_("test")
+
+        def select(query: QueryBuilder) -> QueryBuilder:
+            return query.select("test1", "test2")
+
+        def count_group(query: QueryBuilder, *groups) -> QueryBuilder:
+            return query.groupby(*groups).select(*groups, functions.Count("*"))
+
+        def where_clause(query: QueryBuilder, num_days: int) -> QueryBuilder:
+            return query.where("date" > functions.Now() - num_days)
+
+        for func, args, kwargs in [
+            (select, [], {}),
+            (count_group, ["test1", "test2"], {}),
+            (count_group, ["test1"], {}),
+            (where_clause, [1], {}),
+            (where_clause, [], {"num_days": 1}),
+        ]:
+            self.assertEqual(str(base_query.pipe(func, *args, **kwargs)), str(func(base_query, *args, **kwargs)))
+
+        self.assertEqual(
+            str(base_query.pipe(select).pipe(where_clause, num_days=1)),
+            str(select(where_clause(base_query, num_days=1))),
```

Comment:
I would lean for tests not to rely on any code of the library and simply have the expected string here. Having said that, this can stay but I would also check if it is equal to a desired string of a query. What do you think?

Reply:
Makes sense I will check the expected string